data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/AbstractLogstashPluginAttributesMapper.java

```diff
@@ -33,6 +33,7 @@
         Objects.requireNonNull(logstashAttributesMappings);
         Objects.requireNonNull(logstashAttributesMappings.getMappedAttributeNames());
         Objects.requireNonNull(logstashAttributesMappings.getAdditionalAttributes());
+        Objects.requireNonNull(logstashAttributesMappings.getDefaultSettings());
```

Comment:
This class should later use the default settings to set values which were not previously set. Additionally, this class needs test cases to verify the correct behavior.

Reply:
I see what you mean, more generic for other plugins to use and not tied to OpenSearch. I will update this, thanks.